src/lib/y2storage/partitioning_features.rb

```diff
@@ -0,0 +1,138 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+Yast.import "ProductFeatures"
+
+module Y2Storage
+  # Mixin that allows to load features from partitioning section of control file
+  module PartitioningFeatures
+    # Assign the value of a specific feature into an object attribute.
+    # @param feature [List<Symbol, String>] feature path (e.g., [:proposal, :lvm])
+    # @param to [Symbol, String] attribute name where to store the feature value
+    # @param source [Hash] from where to read the features
+    #
+    # @example
+    #   settings.use_lvm #=> nil
+    #   settings.load_feature(:proposal_lvm, to: :use_lvm)
+    #   settings.use_lvm #=> false
+    #
+    # When :to parameter is not indicated, the last value of feature path is considered
+    # the attribute name:
+    #
+    # @example
+    #   settings.lvm #=> nil
+    #   settings.load_feature(:proposal, :lvm)
+    #   settings.lvm #=> false
+    #
+    # When the feature needs to be loaded from a specific subsection, the subsection can
+    # be passed as :source parameter:
```

Comment:
I got confused by the combination of this text and the example. From the text I was expecting this argument to be the name of the subsection, not its whole content.

Maybe rephrase it like:

By default, the feature is loaded from the content of the `<partitioning>` section of the product features. The argument `source` can be used to provide the content of any other section or subsection.

Reply:
Yes, much better.